File: 品味Java的21種設計模式/dp_java/strategy/example5/DollarCash.java
package cn.javass.dp.strategy.example5;
/**
 * 美元现金支付
 */
public class DollarCash implements PaymentStrategy{
	
	public void pay(PaymentContext ctx) {
		System.out.println("现在给"+ctx.getUserName()+"美元现金支付"+ctx.getMoney()+"元");
	}
}

File: 品味Java的21種設計模式/dp_cpp/strategy/example5/DollarCash.cpp
#include "DollarCash.h"

namespace cn
{
	namespace javass
	{
		namespace dp
		{
			namespace strategy
			{
				namespace example5
				{

					void DollarCash::pay(PaymentContext *ctx)
					{
						puts("现在给"+ctx->getUserName()+"美元现金支付"+ctx->getMoney()+"元");
					}
				}
			}
		}
	}
}